GRDB/Core/Database+Schema.swift

```diff
@@ -365,8 +365,21 @@ extension Database {
                 throw DatabaseError(message: "no such table: \(tableName)")
             }
         }
+        let columnInfoQuery: String
+        if sqlite3_libversion_number() < 3026000 {
+            columnInfoQuery = "PRAGMA table_info(\(tableName.quotedDatabaseIdentifier))"
+        } else {
+            // SQLite 3.26.0 introduced the PRAGMA table_xinfo for finding hidden
+            // columns: https://sqlite.org/releaselog/3_26_0.html
+            // For our purposes, we use this pragma to find generated columns.
+            // The "hidden" column magic numbers come from the SQLite source code.
+            // The values 2 and 3 refer to virtual and stored generated columns,
+            // respectively. Search for COLFLAG_VIRTUAL in
+            // https://www.sqlite.org/cgi/src/file?name=src/pragma.c&ci=fca8dc8b578f215a
+            columnInfoQuery = "SELECT * FROM pragma_table_xinfo('\(tableName)') WHERE hidden IN (0,2,3)"
```

Comment:
Good question. I prefer `hidden IN (0, 2, 3)` as a defensive query because we know what those values mean. On the other hand, we can't predict what `hidden > 3` might select at some point in the future.